ngraph/core/reference/include/ngraph/runtime/reference/transpose.hpp

```diff
@@ -0,0 +1,56 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cmath>
+#include <ngraph/runtime/opt_kernel/reshape.hpp>
+
+#include "ngraph/axis_vector.hpp"
+#include "ngraph/check.hpp"
+#include "ngraph/coordinate_transform.hpp"
+#include "ngraph/op/transpose.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void transpose(const char* data,
+                           const Shape& data_shape,
+                           size_t elem_size,
+                           const T* axis_order,
```

Comment:
You can remove template argument if change a type of axis_order to AxisVector

Reply:
but I think it almost doesn't matter whether to use template function `get_vector` or template `transpose `(performance, binary size). using const T * axis_order argument is more convenient than AxisVector because you don't need to care about the casting when calling the transpose, casting is already inside the function.

Reply:
transpose related changes were reverted